File: 2486-most-frequent-even-element/2486-most-frequent-even-element.py
class Solution:
    def mostFrequentEven(self, nums: List[int]) -> int:
        dic = dict()
        for num in set(nums):
            if num % 2 == 0:
                dic[num] = nums.count(num)
        if len(dic) != 0:
            sorteddic = dict(sorted(dic.items(), key = lambda x:(-x[1], x[0])))
            print(sorteddic)
            for key, val in sorteddic.items():
                return key
        return -1
        

File: 2486-most-frequent-even-element/2486-most-frequent-even-element.java
class Solution {
    public int mostFrequentEven(int[] nums) {
        Map<Integer, Integer> map = new HashMap<>();
        List<Integer> list = new ArrayList<>();
        int val = 0;
        for(int num: nums){
            if(num % 2== 0) {
                map.putIfAbsent(num, 0);
                int temp = map.get(num) + 1;
                val = Math.max(temp, val);
                map.put(num, temp);
                list.add(num);
            }
        }
        
        Collections.sort(list);

        for(int key : list){
            if(map.get(key) == val){
                return key;
            }
        }
        return -1;  
    }
}